src/sdk/modules/smartSessions.ts

```diff
@@ -1,147 +1,148 @@
-import { type Address, type Hex, encodeAbiParameters } from "viem"
-import { UniActionPolicyAbi } from "../__contracts/abi"
-import { type AnyReferenceValue, parseReferenceValue } from "./utils/Helper"
-
-export type Rule = {
-  /**
-   * EQUAL = 0,
-   * GREATER_THAN = 1,
-   * LESS_THAN = 2,
-   * GREATER_THAN_OR_EQUAL = 3,
-   * LESS_THAN_OR_EQUAL = 4,
-   * NOT_EQUAL = 5
-   */
-  condition: ParamCondition
-  /**
-   * The offset in the calldata where the value to be checked is located.
-   * The offset is in multiples of 32 bytes. (Note: not the offsetIndex)
-   * The offsetIndex is generally the index of the arg in the method that you wish to target.
-   * The exception is when the arg is in an array
-   * In this case, the offsetIndex needs to be figured out using its position in the array
-   * (See the 'use-of-dynamic-types' example below for how to figure out the offsetIndex for an array)
-   *
-   * https://docs.soliditylang.org/en/develop/abi-spec.html#use-of-dynamic-types
-   *
-   * */
-  offsetIndex: number
-  /**
-   * If the rule is limited, the usage object will contain the limit and the used values.
-   */
-  isLimited: boolean
-  /**
-   * The reference value to compare against. You can pass in the raw hex value or a human-friendly value.
-   * Use the raw hex value if you are sure of the value you are passing in.
-   */
-  ref: AnyReferenceValue
-  /**
-   * The usage object will contain the limit and the used values, and is only required if the isLimited property is true.
-   */
-  usage: LimitUsage
-}
-
-export type ActionConfig = {
-  valueLimitPerUse: bigint
-  paramRules: {
-    length: number
-    rules: Rule[]
-  }
-}
-
-export const toActionConfig = (config: ActionConfig): RawActionConfig => ({
```

Comment:
I thought this was useful. It made things easier for the end user, using `offsetIndex` and `parseReferenceValue`. Otherwise users have to figure it out themselves which is worse devEx

Reply:
it will just be moved elsewhere once fully clear will bring it back

Reply:
sounds good

Reply:
this is done